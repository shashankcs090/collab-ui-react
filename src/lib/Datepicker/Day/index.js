import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@collab-ui/react';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Day extends React.Component {
  static displayName = 'Day';

  handleClick = e => {
    const { handleDayClick } = this.context;
    const { day } = this.props;
    return (
      handleDayClick
      && handleDayClick(day, e)
    );
  };

  render() {
    const { day, month } = this.props;
    const { selected, focus } = this.context;

    const isOutsideMonth = month !== undefined && month !== utils.getMonth(day);
    const isSelected = utils.isSameDay(day, selected);
    const isToday = utils.isSameDay(day, utils.newDate());
    const disabled = utils.isDayDisabled(day, this.props);
    const hasFocus = utils.isSameDay(day, focus);

    return (
      <Button
        circle
        disabled={disabled}
        className={
          'cui-datepicker__day' +
          `${(isSelected && ` cui-datepicker__day--selected`) || ''}` +
          `${(hasFocus && ` cui-datepicker__day--focus`) || ''}` +
          `${(isToday && ` cui-datepicker__day--today`) || ''}` +
          `${(isOutsideMonth && ` cui-datepicker__day--outside-month`) || ''}`
        }
        onClick={this.handleClick}
        ariaLabel={`day-${utils.getDate(day)}`}
        aria-selected={isSelected}
        tabIndex={-1}
      >
        {utils.getDate(day)}
      </Button>
    );
  }
}

Day.contextTypes = {
  handleDayClick: PropTypes.func,
  focus: PropTypes.object,
  selected: PropTypes.object,
};

Day.propTypes = {
  day: PropTypes.object,
  month: PropTypes.number,
};
