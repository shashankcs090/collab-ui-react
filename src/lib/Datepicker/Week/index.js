import React from 'react';
import PropTypes from 'prop-types';
import Day from '@collab-ui/react/Datepicker/Day';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Week extends React.PureComponent {
  static displayName = 'Week';

  render() {
    const {
      day,
      ...otherProps
    } = this.props;

    const renderDays = () => {
      const startOfWeek = utils.getStartOfWeek(utils.cloneDate(day));

      const days = [0, 1, 2, 3, 4, 5, 6].map(offset => {
        const day = utils.addDays(utils.cloneDate(startOfWeek), offset);
        return (
          <Day
            key={offset}
            day={day}
            {...otherProps}
          />
        );
      });

      return days;
    };

    return (
      <div className="cui-datepicker__week">
        {renderDays()}
      </div>
    );
  }
}

Week.propTypes = {
  day: PropTypes.object.isRequired,
};
