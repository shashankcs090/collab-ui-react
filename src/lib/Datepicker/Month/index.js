import React from 'react';
import PropTypes from 'prop-types';
import Week from '@collab-ui/react/Datepicker/Week';
import * as utils from '@collab-ui/react/Datepicker/date_utils';

export default class Month extends React.Component {
  static displayName = 'Month';

  renderWeeks = () => {
    const { day, ...otherProps } = this.props;

    let i = 0;
    let currentWeekStart = utils.getStartOfWeek(
      utils.getStartOfMonth(utils.cloneDate(day))
    );

    const weeks = [];
    const month = utils.getMonth(day);

    do {
      weeks.push(
        <Week
          key={i++}
          day={currentWeekStart}
          month={month}
          {...otherProps}
        />
      );
      currentWeekStart = utils.addWeeks(utils.cloneDate(currentWeekStart), 1);

    } while(utils.isSameMonth(currentWeekStart, day));

    return weeks;
  };


  render() {
    return (
      <div
        className='cui-datepicker__month'
      >
        {this.renderWeeks()}
      </div>
    );
  }
}

Month.propTypes = {
  day: PropTypes.object.isRequired,
};
