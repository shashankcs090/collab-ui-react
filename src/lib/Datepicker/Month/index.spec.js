import React from 'react';
import { shallow, mount } from 'enzyme';
import Month from '../Month';
import Week from '../Week';
import Day from '../Day';
import moment from 'moment-timezone';

const TEST_DATE = '2018-04-01';

describe('tests for <Month />', () => {
  let day;

  beforeAll(() => {
    // setting timezone to CST
    moment.tz.setDefault('Asia/Kolkata');
    day = moment(TEST_DATE);
  });

  afterAll(() => {
    moment.tz.setDefault();
  });

  it('should match month SnapShot', () => {
    const container = shallow(
      <Month day={day} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render all days of the month correctly', () => {
    const container = mount(
      <Month day={day} />
    );
    expect(container.find(Week)).toHaveLength(5);
    expect(container.find(Day)).toHaveLength(35);
  });
});
