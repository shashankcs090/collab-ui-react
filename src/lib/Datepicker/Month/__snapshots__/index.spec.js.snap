// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <Month /> should match month SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Month
    day={"2018-03-31T18:30:00.000Z"}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Week
          day={"2018-03-31T18:30:00.000Z"}
          month={3}
        />,
        <Week
          day={"2018-04-07T18:30:00.000Z"}
          month={3}
        />,
        <Week
          day={"2018-04-14T18:30:00.000Z"}
          month={3}
        />,
        <Week
          day={"2018-04-21T18:30:00.000Z"}
          month={3}
        />,
        <Week
          day={"2018-04-28T18:30:00.000Z"}
          month={3}
        />,
      ],
      "className": "cui-datepicker__month",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "day": "2018-03-31T18:30:00.000Z",
          "month": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-07T18:30:00.000Z",
          "month": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-14T18:30:00.000Z",
          "month": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-21T18:30:00.000Z",
          "month": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "day": "2018-04-28T18:30:00.000Z",
          "month": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Week
            day={"2018-03-31T18:30:00.000Z"}
            month={3}
          />,
          <Week
            day={"2018-04-07T18:30:00.000Z"}
            month={3}
          />,
          <Week
            day={"2018-04-14T18:30:00.000Z"}
            month={3}
          />,
          <Week
            day={"2018-04-21T18:30:00.000Z"}
            month={3}
          />,
          <Week
            day={"2018-04-28T18:30:00.000Z"}
            month={3}
          />,
        ],
        "className": "cui-datepicker__month",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "day": "2018-03-31T18:30:00.000Z",
            "month": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-07T18:30:00.000Z",
            "month": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-14T18:30:00.000Z",
            "month": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-21T18:30:00.000Z",
            "month": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "day": "2018-04-28T18:30:00.000Z",
            "month": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
